themes/bootstrap3/js/list_item_selection.js

```diff
@@ -235,8 +242,27 @@ VuFind.register("listItemSelection", function ListItemSelection() {
       'checkedDefault': checkedDefault,
     });
     _updateSelectionState(form);
+    boundFunction = _writeState.bind(this, form);
+    window.addEventListener('beforeunload', boundFunction);
+  }
 
-    window.addEventListener('beforeunload', () => _writeState(form));
+  /**
+   * Clear selected items from session storage when delete is called.
+   *
+   * @param {SubmitEvent} event 
+   * @param {object} data 
+   * @returns 
+   */
+  function onDeleteSubmitHandler(event, data) {
+    let form = 'form-favorites';
+    const formData = data.find(el => el.name === 'listID');
+    if (formData) {
+      form += '-' + formData.value;
+    }
+    window.sessionStorage.setItem(form, '{}');
```

Comment:
In my opinion this is a rather brute force approach. Would it be safer to uncheck selected items on the current page, remove the selected items from the `non_default_ids` and then calling `_updateSelectionState(form);` and `_writeState(form);` ? With this approach this could also be reused for the single delete button.

Removing the boundFunction from the eventListeners only works because the site is refreshed after closing the lightbox. But that behavior could change in the future.

I am not completely against your approach but these are things that we have to keep in mind.

Reply:
Yea, I'll try to see and adjust the functionality.